enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,703 @@
+  - The scheduling gate must be removed from the pods manually
+
+### Drawbacks
+- Each pod going through this workflow would report an increased time to schedule, given that image inspection requires registry access. When this is done for a large number of pods, there might be a considerable performance impact (to be determined)
+- The controller would need access to potentially every secret configured in the cluster to function deterministically. 
+- If the operator is uninstalled abruptly when pods are going through the interception and schedule gating phase, the operator would have to be installed again for these pods to be unblocked and scheduled, or these pods would need to be patched to remove the scheduling gate to continue to be scheduled as normal. 
+
+## Design Details
+
+### Deployment model/diagram
+The operator will deploy the different controllers as separate pods. In particular, the WebHook and the PodPlacementController have to be considered separate deployments due to their diverse nature: the former is a stateless service, and the latter will be deployed in HA following the primary-backup pattern.
+
+The following diagram reflects the proposed deployment model.
+
+```mermaid
+%%{init: {'securityLevel': 'loose', 'theme':'neutral', 'themeVariables': { 'fontSize':'14px', 'fontFamily':'Roboto Mono'}}}%%
+
+classDiagram
+    direction LR
+
+    class PodPlacementController {
+        <<Deployment>>
+        replicas=2
+    }
+
+    class Operator {
+        <<Deployment>>
+        <<primary-backup>>
+        replicas=2
+
+    }
+
+    class SchedulingGateMutatingWebhookConfiguration {
+        <<MutatingWebhookConfiguration>>
+        operations = [CREATE]
+        sideEffects = None
+        failurePolicy = Ignore
+    }
+
+    class PlacementPodConfig {
+        <<CR>>
+
+    }
+
+    class Pod {
+        schedulingGates: List
+        affinity: Object
+    }
+
+    class SchedulingGateMutatingWebhook {
+        <<Deployment>>
+        replicas=3
+    }
+    SchedulingGateMutatingWebhookSvc --> SchedulingGateMutatingWebhook 
+
+    class SchedulingGateMutatingWebhookSvc {
+        <<service>>
+    }
+
+    SchedulingGateMutatingWebhookConfiguration --> SchedulingGateMutatingWebhookSvc
+    Operator --> PodPlacementController : creates and reconciles
+    Operator --> SchedulingGateMutatingWebhook : creates and reconciles
+    Operator --> PlacementPodConfig : reconciles
+    PodPlacementController --> Pod : updates (patches nodeAffinity,<br/>removes schedulingGate)
+    SchedulingGateMutatingWebhook --> Pod : updates (adds schedulingGate)
+
+```
+### Open Questions
+- What is the performance impact when pods are gated and inspected over many
+  namespaces? This needs to be qualified. Mitigations have already been
+  discussed in the other sections, but we expect others to address at
+  implementation and performance evaluation time.
+- Should the functionality ignore Openshift namespaces (openshift-\*) by
```

Comment:
The proposed feature should be user workloads only focused. The components under openshift-*, hypershift-* and other openshift dedicated namespaces are owned by OpenShift teams and we know which architectures are supported. By default every architecture is supported unless explicitly state otherwise. Plus, we don't want either of our pods to be mutated in a way that might make the pods unschedulable. Any such situation can potential put the openshift component in risk. The control plane components (etcd, KA, KCM, KS) are static pods so there's no way to mutate those. Nevertheless, openshift-apiserver, oauth-openshift, openshift-controller-manager are not and when improperly mutated they can lead to instability or malfunction of the OpenShift API.